packages/Confirmation/src/Confirmation.js

```diff
@@ -0,0 +1,158 @@
+import React from "react";
+import PropTypes from "prop-types";
+import uuid from "uuid/v4";
+import Button from "@paprika/button";
+import Heading from "@paprika/heading";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import Popover from "@paprika/popover";
+import Trigger from "./components/Trigger";
+
+import { confirmStyles, confirmBodyStyles, confirmFooterStyles } from "./Confirmation.styles";
+
+const propTypes = {
+  buttonSize: PropTypes.oneOf(ShirtSizes.DEFAULT),
+  /** Children should be a render prop in the form of a function to display trigger */
+  children: PropTypes.func,
+  confirmButtonType: PropTypes.oneOf(["primary", "destructive"]),
+  confirmLabel: PropTypes.string.isRequired,
+  body: PropTypes.node,
+  heading: PropTypes.string,
+  defaultIsOpen: PropTypes.bool,
+  isPending: PropTypes.bool,
+  onClose: PropTypes.func,
+  onConfirm: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  buttonSize: ShirtSizes.MEDIUM,
+  children: null,
+  confirmButtonType: "destructive",
+  defaultIsOpen: false,
+  body: null,
+  heading: null,
+  isPending: false,
+  onClose: () => {},
+};
+
+const Confirmation = props => {
+  const [isConfirmOpen, setIsConfirmOpen] = React.useState(false);
+  const {
+    heading,
+    buttonSize,
+    children,
+    confirmButtonType,
+    confirmLabel,
+    body,
+    defaultIsOpen,
+    isPending,
+    onConfirm,
+    onClose,
+    ...moreProps
+  } = props;
+  const confirmButtonRef = React.useRef(null);
+  const triggerRef = React.useRef(null);
+  const confirmId = React.useRef(uuid());
+
+  const focusConfirmButton = () => {
+    if (confirmButtonRef.current) confirmButtonRef.current.focus();
+  };
+
+  const handleOpenConfirm = () => {
+    setIsConfirmOpen(true);
+    setTimeout(() => {
+      focusConfirmButton();
+    }, 200);
+  };
+
+  React.useEffect(() => {
+    if (defaultIsOpen && !isConfirmOpen) {
+      handleOpenConfirm();
+    }
+  }, [confirmButtonRef, defaultIsOpen]);
+
+  const triggerProps = {
+    isConfirmOpen,
+    handleOpenConfirm,
+  };
+
+  const closeConfirm = () => {
+    setIsConfirmOpen(false);
+  };
+
+  const handleOnConfirm = () => {
+    onConfirm(closeConfirm);
```

Comment:
```suggestion
    onConfirm(handleCloseConfirm);
```

Reply:
Oh, you'll have to do a little more work after this change to appease the linter... 😬